public/app/features/explore/state/reducers.ts

```diff
@@ -511,7 +511,11 @@ export const processQueryResponse = (
 
   if (error) {
     if (error.cancelled) {
-      return state;
+      return {
```

Comment:
Not 100% sure that this is okay to do, but seemed like the simplest way to be able to keep fetched results while still displaying an error message

Reply:
Hmm not sure about this. If I get this correctly this works for states when it times out but still got some response data. So like CloudWatch logs where some data can come through even though some may timeout.

For other datsources this will probably do the opposite. If the response times out before we preserved any previous data but now we will replace it with the response that probably does not have any data, is that right?

Reply:
Ah, you're right, seems like some more work will have to go into this. What do you think of adding a different `DataQueryError` prop like `timedOut`, or perhaps better, refactoring the interface so that there's an `errorType` prop that could be one of `ErrorType.Cancelled`, `ErrorType.TimeOut` etc

Reply:
So checking now and it seems like in case of errors we do not keep the previous response data anyway. I mean it may be confusing. Right now you can get response with data + error like in this case and it would be confusing if there were cases where we left the previous data cause than you wouldn't know if that is old data or you get new response but it's looking the same. So I think this approach is fine.